lib/rails_param/coercion/hash_param.rb

```diff
@@ -0,0 +1,19 @@
+module RailsParam
+  module Param
+    class Coercion
+      class HashParam
+        attr_reader :param, :options, :type
+
+        def initialize(param:, options: nil, type: nil)
+          @param, @options, @type = param, options, type
+        end
+
+        def coerce
+          raise ArgumentError unless param.respond_to?(:split)
+
+          Hash[param.split(options[:delimiter] || ",").map { |c| c.split(options[:separator] || ":") }]
```

Comment:
Oof I know this is how it was originally written but I don't think this would play well with nested hashes. Definitely something that could be improved upon later.